disnake/shard.py

```diff
@@ -426,28 +427,36 @@ async def launch_shard(self, gateway: str, shard_id: int, *, initial: bool = Fal
         self.__shards[shard_id] = ret = Shard(ws, self, self.__queue.put_nowait)
         ret.launch()
 
-    async def launch_shards(self) -> None:
+    async def launch_shards(self, *, ignore_session_start_limit: bool = False) -> None:
         shard_count, gateway, session_start_limit = await self.http.get_bot_gateway()
-        if self.shard_count is None:
-            self.shard_count = shard_count
 
         self.session_start_limit = SessionStartLimit(session_start_limit)
 
+        if self.shard_count is None:
+            self.shard_count = shard_count
+
         self._connection.shard_count = self.shard_count
 
         shard_ids = self.shard_ids or range(self.shard_count)
         self._connection.shard_ids = shard_ids
 
+        if not ignore_session_start_limit and self.session_start_limit.remaining < self.shard_count:
+            raise SessionStartLimitReached(
+                f"Daily session start limit has been reached, resets at {self.session_start_limit.reset_time}"
```

Comment:
Might be useful to include the requested and remaining number of shards in the error message

Reply:
https://github.com/DisnakeDev/disnake/pull/537/commits/126f9b89c9c0d8443100b673215c472da34c8e03